src/Link.cc

```diff
@@ -304,4 +308,46 @@ namespace kickcat
         uint16_t wkc = wkc_nominal + wkc_redundancy;
         return std::make_tuple(header_nominal, data_nominal, wkc);
     }
+
+
+    void Link::attachEcatEventCallback(enum EcatEvent ecat_event, std::function<void()> callback)
+    {
+        int32_t index = 0;
+        uint16_t mask = ecat_event;
+        while (mask != 0)
+        {
+            if (mask & 1)
+            {
+                auto& event = irqs_[index];
+                event.callback = callback;
+            }
+
+            ++index;
+            mask = mask >> 1;
+        }
+    }
+
+
+    void Link::checkEcatEvents(uint16_t irq)
+    {
+        for (uint16_t i = 0; i < irqs_.size(); ++i)
+        {
+            auto& event = irqs_[i];
+            if (irq & 1)
+            {
+                if (event.is_armed)
+                {
+                    event.is_armed = false;     // Disable event: we want to trigger on slope
+                    event.callback();
```

Comment:
Calling an empty std::function() will crash the application. It shall never happen by construction (IRQ are resetted at init) but still, if a slave is buggy, it may crash the master... I add a default no-op handler to avoid a bad surprise.